src/ui/ModelProperty.cpp

```diff
@@ -0,0 +1,41 @@
+#include "ModelProperty.hh"
```

Comment:
ModelProperties are largely based on WPF DependencyProperties. A property has a unique identifier and a default value. A map in the ViewModel hashes the unique identifier to the current value of the property. If the value is not in the hash, the default value is returned.

The primary reason for using hashes of values is for binding the ViewModel to the UI for automatic synchronizing (not needed for MessageBoxes), but this PR sets things up for that. I plan to do the Rich Presence monitor next to illustrate simple binding before moving onto the Login dialog to illustrate control binding.